File: backend/app/schemas/dataset.py
from datetime import datetime, timezone

from pydantic import BaseModel, Field


class HealthResponse(BaseModel):
    status: str = "ok"
    service: str = "dataforge-ai"
    version: str = "0.1.0"
    timestamp: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class ColumnProfile(BaseModel):
    name: str
    dtype: str
    non_null_count: int
    null_count: int
    null_pct: float
    unique_count: int
    sample_values: list[str] = Field(default_factory=list)


class DatasetProfile(BaseModel):
    dataset_id: str
    filename: str
    row_count: int
    column_count: int
    columns: list[ColumnProfile]
    memory_mb: float


class DatasetUploadResponse(BaseModel):
    dataset_id: str
    filename: str
    message: str
    profile: DatasetProfile

File: frontend/src/types/dataset.ts
export interface ColumnProfile {
  name: string;
  dtype: string;
  non_null_count: number;
  null_count: number;
  null_pct: number;
  unique_count: number;
  sample_values: string[];
}

export interface DatasetProfile {
  dataset_id: string;
  filename: string;
  row_count: number;
  column_count: number;
  columns: ColumnProfile[];
  memory_mb: number;
}


export interface DatasetUploadResponse {
  dataset_id: string;
  filename: string;
  message: string;
  profile: DatasetProfile;
}

export interface HealthResponse {
  status: string;
  service: string;
  version: string;
  timestamp: string;
}

export interface PipelineStep{
  step_id:string;
  issue_type: string;
  target_column?: string;
  action_taken: string;
  timestamp: string;
  affected_rows_count: number;
  parameters_used: Record<string, any>;
}

export interface DatasetVersionNode{
  version_id: number;
  parent_version_id: number | null;
  file_path: string;
  created_at: string;
  applied_step: PipelineStep | null;
}

export interface VersionManifest {
  dataset_id: string;
  current_version: number;
  versions: DatasetVersionNode[];
}

